gcm.go

```diff
@@ -194,6 +195,12 @@ func (c *httpGcmClient) send(apiKey string, m HttpMessage) (*HttpResponse, error
 		return nil, fmt.Errorf("error sending request to HTTP connection server>%v", err)
 	}
 	gcmResp := &HttpResponse{}
+	gcmResp.Status = httpResp.StatusCode
+
+	if gcmResp.Status != http.StatusOK {
+		return gcmResp, fmt.Errorf("Encountered HTTP status code %d", gcmResp.Status)
```

Comment:
@jpatel531 I don't think the upstream client is really interested in the response status code and its meaning, it needs either a response object or an error explaining why there's no object. So in case of http statuses 400,401,5xx in the spec, it would be something along the lines of

```go
gcmResp := &HttpResponse{}
body, err := ioutil.ReadAll(httpResp.Body)
defer httpResp.Body.Close()
if err != nil {
        return gcmResp, fmt.Errorf("error reading http response body: %v", err)
}
if httpResp.StatusCode >= http.StatusBadRequest { // Not sure about 3xx 
        return nil, fmt.Errorf("http error: " + httpResp.Status)
}
err = json.Unmarshal(body, &gcmResp)
if err != nil {
        return nil, fmt.Errorf("error unmarshaling json from body: %v %s", err, string(body))
}
```